test/decorator.test.js

```diff
@@ -60,6 +60,27 @@ test('decorate should throw if a declared dependency is not present', t => {
   fastify.ready(() => t.pass())
 })
 
+// issue #777
+test('should throw for missing request decorator', t => {
+  t.plan(2)
+  const fastify = Fastify()
+
+  try {
+    const plugin = fp(function (instance, opts, next) {
+      next()
+    }, {
+      decorators: {
+        request: ['foo']
+      }
+    })
+    fastify
+      .register(plugin)
+      .ready(t.pass)
```

Comment:
the error should land into the ready callback, not being thrown.

This a probably a bug here and not in avvio.

Reply:
So are we saying this pattern is invalid? Even if so, this failed assert still results in an `UncaughtException` due to the nature of the way plugins are registered via `avvio`:

https://github.com/fastify/fastify/blob/81b6e01a73f9022e64b18fd5cc089786dbda87ee/lib/pluginUtils.js#L46-L49

Reply:
The pattern is not valid, the test should verify that an error gets passed to ready. Currently it’s not.